### DISCUSSION TITLE
Checkout breaks when hiding delivery options with Type == PICK_UP

### DISCUSSION THREAD

**Discussion by gil--** [Shopify/function-examples#424] [Feedback: Delivery Customization API]
## Issue

If we leverage a delivery customization function to hide any [delivery option](https://shopify.dev/docs/api/functions/reference/delivery-customization/graphql/common-objects/deliverymethod) with type === pickup, it will hide the pickup locations but not the delivery type resulting in a half broken checkout:

![CleanShot 2024-01-12 at 12 37 24@2x](https://github.com/Shopify/function-examples/assets/3484527/484a8370-585b-41d2-9fcc-973e61d73ab3)

This seems like a bug/undesired state for a customer in checkout.

## Expected result

The pickup option is hidden from customers and checkout behaves as if no pickup options exist.

### Checkout
https://github.com/Shopify/function-examples/assets/3484527/510151d3-87cc-4438-a9b0-883e4b5ebf25

### Shop Pay
https://github.com/Shopify/function-examples/assets/3484527/d8dfce20-f935-4c79-9186-0acf16788051

**Comment by rbarreca** [comment]
Related: https://github.com/Shopify/function-examples/discussions/218

**Comment by KevenLYF** [comment]
Hi, we recently shipped a fix around that area, could you give it a try again ? Thanks

**Comment by gil--** [comment]
confirmed fixed on my end.
